File: examples/03-interactions/go/workflow_and_run_timeouts.go
package interactions

import (
	"context"
	"time"

	"go.temporal.io/sdk/client"
)

// Two distinct lifetimes on the start options:
//
//	WorkflowExecutionTimeout - the whole Workflow, summed across continue-as-new.
//	WorkflowRunTimeout       - one individual run before it must finish or CAN.
func Start(ctx context.Context, c client.Client) error {
	_, err := c.ExecuteWorkflow(ctx, client.StartWorkflowOptions{
		ID:        "orders-2026-05-27",
		TaskQueue: "orders",
		// Maximum lifetime across continue-as-new runs.
		WorkflowExecutionTimeout: 7 * 24 * time.Hour,
		// Maximum lifetime for this individual run.
		WorkflowRunTimeout: 12 * time.Hour,
	}, OrdersWorkflow, "2026-05-27")
	return err
}

File: examples/03-interactions/python/workflow_and_run_timeouts.py
from datetime import timedelta

from temporalio.client import Client


# Two distinct lifetimes on the start options:
#   execution_timeout - the whole Workflow, summed across continue-as-new runs.
#   run_timeout       - one individual run before it must finish or continue-as-new.
async def start(client: Client) -> None:
    await client.start_workflow(
        OrdersWorkflow.run,
        "2026-05-27",
        id="orders-2026-05-27",
        task_queue="orders",
        # Maximum lifetime across continue-as-new runs.
        execution_timeout=timedelta(days=7),
        # Maximum lifetime for this individual run.
        run_timeout=timedelta(hours=12),
    )
